### PR TITLE
Improve things arround change detection

### PR DESCRIPTION
**Pull request by tobiasdiez**
Another shot at fixing https://github.com/JabRef/jabref/issues/4877. The changes in this PR might be already sufficient, let's see...

No time for extended PR description, see commit messages for changes. Last commit contains the actual "fix", the other ones are cosmetic changes.

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not: Issue created at <https://github.com/JabRef/user-documentation/issues>.

### CODE REVIEW

File: src/main/java/org/jabref/logic/importer/OpenDatabase.java [hunk 1]
```diff
@@ -62,9 +59,9 @@ public static ParserResult loadDatabase(String name, ImportFormatPreferences imp
     /**
      * Opens a new database.
      */
-    public static ParserResult loadDatabase(File fileToOpen, ImportFormatPreferences importFormatPreferences, FileUpdateMonitor fileMonitor)
+    public static ParserResult loadDatabase(Path fileToOpen, ImportFormatPreferences importFormatPreferences, FileUpdateMonitor fileMonitor)
         throws IOException {
-        ParserResult result = new BibtexImporter(importFormatPreferences, fileMonitor).importDatabase(fileToOpen.toPath(),
+        ParserResult result = new BibtexImporter(importFormatPreferences, fileMonitor).importDatabase(fileToOpen,
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/importer/OpenDatabase.java]: Why did you change it from path to file?

Comment (koppor) [reply src/main/java/org/jabref/logic/importer/OpenDatabase.java]: This question is asked, because java.nio.Path should be preferred over java.io.File. We are aware in the code that this causes some conversion issues, however, I think, we should keep to move forward here. (Is this article still the best google result? --> https://jaxenter.de/java-nio-file-zeitgemases-arbeiten-mit-dateien-2581)

File: src/main/java/org/jabref/logic/importer/OpenDatabase.java [hunk 2]
```diff
@@ -1,7 +1,9 @@
 package org.jabref.logic.importer;
 
-import java.io.File;
 import java.io.IOException;
+import java.nio.file.Files;
```
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/OpenDatabase.java]: @Siedlerchr here is a change to java.nio 🎉

File: src/main/java/org/jabref/logic/importer/OpenDatabase.java [hunk 3]
```diff
@@ -29,31 +31,26 @@ private OpenDatabase() {
      *
      * @param name Name of the BIB-file to open
      * @return ParserResult which never is null
+     * @deprecated use {@link #loadDatabase(Path, ImportFormatPreferences, FileUpdateMonitor)} instead
      */
+    @Deprecated
     public static ParserResult loadDatabase(String name, ImportFormatPreferences importFormatPreferences, FileUpdateMonitor fileMonitor) {
-        File file = new File(name);
-        LOGGER.info("Opening: " + name);
+        LOGGER.debug("Opening: " + name);
```
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/OpenDatabase.java]: Doesn't "Opening {}", name work? - I think, this is the style one does in Java, because otherwise the debug string is always calculated even if the statement is not set.

File: src/main/java/org/jabref/gui/collab/DatabaseChangeMonitor.java [hunk 4]
```diff
@@ -34,9 +30,6 @@ public DatabaseChangeMonitor(BibDatabaseContext database, FileUpdateMonitor file
         this.database.getDatabasePath().ifPresent(path -> {
             try {
                 fileMonitor.addListenerForFile(path, this);
-                referenceFile = Files.createTempFile("jabref", ".bib");
```
Comment (koppor) [inline src/main/java/org/jabref/gui/collab/DatabaseChangeMonitor.java]: I like that removal - even though I think, one still needs that reference somehow. Maybe only for the approach sketched at https://github.com/JabRef/jabref/pull/5669.

File: src/main/java/org/jabref/logic/importer/OpenDatabase.java (hunk 3 above)
Comment (tobiasdiez) [reply src/main/java/org/jabref/logic/importer/OpenDatabase.java]: Isn't that micro optimization?

File: src/main/java/org/jabref/logic/importer/OpenDatabase.java (hunk 1 above)
Comment (tobiasdiez) [reply src/main/java/org/jabref/logic/importer/OpenDatabase.java]: I actually converted it to a Path, the method now accepts a Path variable and not a File

**Comment by tobiasdiez** [comment]
I agree writing tests would be a good idea. However that requires major refactoring beforehand as the problematic code is still tightly coupled to the UI 😣
